mod/messageboard/start.php

```diff
@@ -11,8 +11,9 @@
  * MessageBoard initialisation
  */
 function messageboard_init() {
-	// js
-	elgg_extend_view('elgg.js', 'messageboard/js');
+	// inline module
+	elgg_extend_view('elgg.js', 'elgg/messageboard.js');
+	elgg_require_js('elgg/messageboard');
```

Comment:
As you say, we will have to support this through 3.x. I guess the impact is minimal butwe would have to keep the module available on all requests